src/api/tests/test_llm.py

```
@@ -460,6 +460,58 @@ def fake_completion(*args, **kwargs):
+
+    assert params["allowed_openai_params"] == ["reasoning_effort"]
+    assert params["reasoning_effort"] == "low"
+
+
+def test_chat_llm_ends_partial_response_on_repeated_stream_chunk(monkeypatch, app):
+    class RepeatedChunkError(Exception):
+        __module__ = "litellm.exceptions"
+
+    def fake_completion(*args, **kwargs):
+        yield FakeResponse("chunk-1", content="你好")
+        raise RepeatedChunkError("The model is repeating the same chunk = ！ ！ .")
```

Ruff (0.15.12)

[warning] String contains ambiguous `！` (FULLWIDTH EXCLAMATION MARK). Did you mean `!` (EXCLAMATION MARK)?

(RUF001)